x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MountSnapshotStep.java

```diff
@@ -101,10 +103,14 @@ void performDuringNoSnapshot(IndexMetadata indexMetadata, ClusterState currentCl
             indexName = snapshotIndexName;
         }
 
+        Settings.Builder settingsBuilder = Settings.builder();
+        settingsBuilder.put(IndexSettings.INDEX_CHECK_ON_STARTUP.getKey(), Boolean.FALSE.toString());
+        if (TimeseriesLifecycleType.HOT_PHASE.equals(this.getKey().getPhase())) {
```

Comment:
Can you add a comment about why this is necessary here?

Reply:
Sure thing, 21cabd43d71dd6e77c4d9c1729f05fbe3c7c75f9